### PR TITLE
various changes to WorkdayReports raw_reports class

### PR DESCRIPTION
**Pull request by mozafrank**
This PR adds some functionality to WorkdayReports, notably:
- read and write files
- parse the csv
- add __iter__
- added some help() text

### CODE REVIEW

File: WorkdayReports/raw_reports.py [hunk 1]
```diff
@@ -1,40 +1,124 @@
...
+import io
+import csv
 
 logger = logging.getLogger(__name__)
 
+class Report:
+    """
+    This class can download a report based on URI from workday (or read it from file), and optionally parse the report, store it, or return it raw.
+    Parameters:
+    * report_url, base_url, report_uri can be used to define where the report is retrieved from. report_url takes precedent over base_url+report_uri. A sane base_url is provided. Either report_url or report_uri are required unless using input_filename, which will forego hitting workday.
```
Comment (claudiouzelac) [inline WorkdayReports/raw_reports.py]: What is the column length that this codebase generally adheres too?  I ask as these lines are quite long.

**Review by claudiouzelac — APPROVED** [review]
LGTM; minor comment added in the PR

File: WorkdayReports/raw_reports.py (hunk 1 above)
Comment (mozafrank) [reply WorkdayReports/raw_reports.py]: yeah sorry about that, undoubtedly a result of my nutty desktop setup; @bowlofstew do you have a recommended line length? We could just say 80 or whatever the default terminal width is.

Comment (bsieber-mozilla) [reply WorkdayReports/raw_reports.py]: I think it's using Black in pre-commit if those were run

Comment (bsieber-mozilla) [reply WorkdayReports/raw_reports.py]: Should be 80 or 87/88 from Black

Comment (mozafrank) [reply WorkdayReports/raw_reports.py]: ah perfect thanks @bsieber-mozilla I'll run black locally and fix.

Comment (mozafrank) [reply WorkdayReports/raw_reports.py]: Formatting updated

File: WorkdayReports/raw_reports.py [hunk 2]
```diff
@@ -1,40 +1,138 @@
...
+            self.read()
+            self.parse()
+            # if we're just reading a file we can stop here.
+            return None
+
+        if not self.report_url and base_url and report_uri:
+            self.report_url = f"{base_url}/{report_uri}"
+        if "?" not in self.report_url:
+            self.report_url += "?format=csv&bom=true"
+        if "format=csv" not in self.report_url:
```
Comment (cvalaas) [inline WorkdayReports/raw_reports.py:67]: I use `format=json` in several reports

**Review by cvalaas — CHANGES_REQUESTED** [review]
JSON is another popular format for getting workday reports. Needed for several report formats

File: WorkdayReports/raw_reports.py (hunk 2 above)
Comment (mozafrank) [reply WorkdayReports/raw_reports.py:67]: I talked to @cvalaas about this and I'm going to address json input support in a subsequent PR, so I'm resolving this for now.

**Pull request merged**